components/suggest/src/yelp.rs

```diff
@@ -337,27 +376,29 @@ impl<'a> SuggestDao<'a> {
 }
 
 struct SuggestionBuilder<'a> {
-    query: &'a SuggestionQuery,
     subject: &'a str,
+    subject_exact_match: bool,
     pre_modifier: Option<String>,
     post_modifier: Option<String>,
     location_sign: Option<String>,
     location: Option<String>,
     need_location: bool,
+    pre_yelp_modifier: Option<String>,
+    post_yelp_modifier: Option<String>,
 }
 
 impl<'a> From<SuggestionBuilder<'a>> for Suggestion {
     fn from(builder: SuggestionBuilder<'a>) -> Suggestion {
         // This location sign such the 'near by' needs to add as a description parameter.
         let location_modifier = if !builder.need_location {
-            builder.location_sign
+            builder.location_sign.clone()
```

Comment:
We can save a couple of clones by this:

```rs
        let location_modifier = if !builder.need_location {
            builder.location_sign.as_deref()
        } else {
            None
        };
        let description = [
            builder.pre_modifier.as_deref(),
            Some(builder.subject),
            builder.post_modifier.as_deref(),
            location_modifier,
        ]
        .iter()
        .flatten()
        .map(|&s| s)
        .collect::<Vec<_>>()
        .join(" ");
```

Reply:
Oh, cool.. I didn't know the `as_deref()`.